File: ide_py_foundation/src/ecc/dennisc/alice/ide/cascade/fillerinners/AngleFillerInner.py
#print "-->", __name__

from ExpressionFillerInner import *

class AngleFillerInner( ExpressionFillerInner ):
	def __init__( self ):
		ExpressionFillerInner.__init__( self, ecc.dennisc.alice.ast.getType( org.alice.apis.moveandturn.AngleInRevolutions ), alice.ast.InstanceCreation )
	def addFillIns( self, blank ):
		cls = org.alice.apis.moveandturn.AngleInRevolutions
		cnstrctr = alice.ast.TypeDeclaredInJava.getConstructor( cls.getConstructor( [java.lang.Number] ) )
		param = cnstrctr.parameters.get( 0 )
		self._addExpressionFillIn( blank, cnstrctr, [ alice.ast.Argument( param, alice.ast.DoubleLiteral( 0.125 ) ) ] ) 
		self._addExpressionFillIn( blank, cnstrctr, [ alice.ast.Argument( param, alice.ast.DoubleLiteral( 0.25 ) ) ] ) 
		self._addExpressionFillIn( blank, cnstrctr, [ alice.ast.Argument( param, alice.ast.DoubleLiteral( 0.5 ) ) ] ) 
		self._addExpressionFillIn( blank, cnstrctr, [ alice.ast.Argument( param, alice.ast.DoubleLiteral( 1.0 ) ) ] ) 
		self._addExpressionFillIn( blank, cnstrctr, [ alice.ast.Argument( param, alice.ast.DoubleLiteral( 2.0 ) ) ] ) 
		self._addExpressionFillIn( blank, cnstrctr, [ alice.ast.Argument( param, alice.ast.DoubleLiteral( 4.0 ) ) ] )
		blank.addSeparator()
		blank.addChild( ecc.dennisc.alice.ide.cascade.CustomAngleFillIn() )
		 

#print "<--", __name__

File: x_ide_stage/src/org/alice/stageide/cascade/fillerinners/AngleFillerInner.java
/*
 * Copyright (c) 2006-2009, Carnegie Mellon University. All rights reserved.
 * 
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are
 * met:
 * 
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 * 
 * 3. Products derived from the software may not be called "Alice",
 *    nor may "Alice" appear in their name, without prior written
 *    permission of Carnegie Mellon University.
 * 
 * 4. All advertising materials mentioning features or use of this software
 *    must display the following acknowledgement:
 *    "This product includes software developed by Carnegie Mellon University"
 */
package org.alice.stageide.cascade.fillerinners;

/**
 * @author Dennis Cosgrove
 */
public class AngleFillerInner extends org.alice.ide.cascade.fillerinners.InstanceCreationFillerInner {
	public AngleFillerInner() {
		super( org.alice.apis.moveandturn.AngleInRevolutions.class );
	}
	@Override
	public void addFillIns( edu.cmu.cs.dennisc.cascade.Blank blank ) {
		edu.cmu.cs.dennisc.alice.ast.AbstractType type = this.getType();
		edu.cmu.cs.dennisc.alice.ast.AbstractConstructor constructor = type.getDeclaredConstructor( new Class<?>[] { Number.class } );
		edu.cmu.cs.dennisc.alice.ast.AbstractParameter parameter = constructor.getParameters().get( 0 );

		this.addInstanceCreationExpressionFillIn( blank, constructor, parameter, new edu.cmu.cs.dennisc.alice.ast.DoubleLiteral( 0.125 ) );
		this.addInstanceCreationExpressionFillIn( blank, constructor, parameter, new edu.cmu.cs.dennisc.alice.ast.DoubleLiteral( 0.25 ) );
		this.addInstanceCreationExpressionFillIn( blank, constructor, parameter, new edu.cmu.cs.dennisc.alice.ast.DoubleLiteral( 0.5 ) );
		this.addInstanceCreationExpressionFillIn( blank, constructor, parameter, new edu.cmu.cs.dennisc.alice.ast.DoubleLiteral( 1.0 ) );
		this.addInstanceCreationExpressionFillIn( blank, constructor, parameter, new edu.cmu.cs.dennisc.alice.ast.DoubleLiteral( 2.0 ) );
		this.addInstanceCreationExpressionFillIn( blank, constructor, parameter, new edu.cmu.cs.dennisc.alice.ast.DoubleLiteral( 4.0 ) );
//		blank.addSeparator();
//		blank.addChild( ecc.dennisc.alice.ide.cascade.CustomAngleFillIn() )
	}
}
